internal/provider/resource_application.go

```diff
@@ -164,6 +165,34 @@ func (r *applicationResource) Schema(_ context.Context, _ resource.SchemaRequest
 					stringplanmodifier.UseStateForUnknown(),
 				},
 			},
+			"storage": schema.SetNestedAttribute{
+				Description: "Configure storage constraints for the juju application.",
+				Optional:    true,
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						"label": schema.StringAttribute{
+							Description: "The specific storage option defined in the charm.",
+							Required:    true,
+						},
+						"size": schema.StringAttribute{
+							Description: "The size of each volume. E.g. 100G",
+							Required:    true,
```

Comment:
From reading the storage constraints docs, it seems to me that size is not required. From what I can tell you can choose to leave out any of size, pool and count. 

I guess it would technically be fine if the user left out all three, it should just have no effect.

Reply:
Hm, and where we said that this is optional... 

Knowing that we do not support dynamic size volumes, I wonder how the "size" could be optional.

Reply:
Find the default that juju uses and make that the default. Then the attribute can be optional.
```
Default: stringdefault.StaticString("str"),
```

I'm not sure if setting a default in the schema is possible or not. Charm metadata can optionally define a MinimumSize for storage:
```
	// There is no default MinimumSize; if left unspecified, a provider
	// specific default will be used, typically 1GB for block storage.
```

Reply:
Here is the doc I was looking at https://juju.is/docs/juju/storage-constraint